cache/network/download_test.go

```diff
@@ -75,3 +86,74 @@ func TestCreateCustomRetryFunction(t *testing.T) {
 		})
 	}
 }
+
+func Test_downloadFile_multipart_retrycheck(t *testing.T) {
+	// Given
+	mockLogger := new(mocks.Logger)
+	mockLogger.On("Debugf", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return()
+
+	retryableHTTPClient := retryhttp.NewClient(mockLogger)
+	var mu sync.RWMutex
```

Comment:
FYI you can use `atomic`'s [bool](https://pkg.go.dev/sync/atomic#Bool.Store)

Reply:
Good point, changed.